cli/glnexus_cli.cc

```diff
@@ -571,6 +572,93 @@ int main_genotype(int argc, char *argv[]) {
     return 0;
 }
 
+int main_iter_compare(int argc, char *argv[]) {
+    if (argc == 2) {
+        help_init(argv[0]);
+        return 1;
+    }
+
+    static struct option long_options[] = {
+        {"help", no_argument, 0, 'h'},
+        {0, 0, 0, 0}
+    };
+
+    int c;
+    optind = 2; // force optind past command positional argument
+    while (-1 != (c = getopt_long(argc, argv, "h",
+                                  long_options, nullptr))) {
+        switch (c) {
+            case 'h':
+            case '?':
+                help_init(argv[0]);
+                exit(1);
+                break;
+
+            default:
+                abort ();
+        }
+    }
+
+//    if (optind != argc-2) {
+//        help_init(argv[0]);
+//        return 1;
+//    }
+    string dbpath(argv[optind]);
+
+    // open the database in read-write mode, create an all-samples sample set,
+    // and close it. This is not elegant. It'd be better for the bulk load
+    // process to create the sample set when it finishes. Need some way to
+    // recover the name of the most recently created all-samples sample set.
+    unique_ptr<GLnexus::KeyValue::DB> db;
+    unique_ptr<GLnexus::BCFKeyValueData> data;
+    string sampleset;
+    H("open database R/W", GLnexus::RocksKeyValue::Open(dbpath, db));
+    H("open database", GLnexus::BCFKeyValueData::Open(db.get(), data));
+
+    unique_ptr<GLnexus::MetadataCache> metadata;
+    H("instantiate metadata cache", GLnexus::MetadataCache::Start(*data, metadata));
+
+    H("all_samples_sampleset", data->all_samples_sampleset(sampleset));
+    console->info() << "created sample set " << sampleset;
+
+    // resolve the user-supplied contig name to rid
+    const auto& contigs = metadata->contigs();
+
+    // get samples and datasets
+    shared_ptr<const set<string>> samples, datasets;
+    H("sampleset_datasets", metadata->sampleset_datasets(sampleset, samples, datasets));
+
+    int nChroms = min((size_t)22, contigs.size());
+    int nIter = 20;
```

Comment:
20 comparisons probably runs in just a few seconds, right? There's enough overhead in starting the applet that it would probably make sense to do a larger number that'll run for a few minutes.